units/projectile.gd

```diff
@@ -1,6 +1,41 @@
 extends Area2D
 
 const SPEED = 750
+var projectile_effects: Array = []
+
+
+func spawn(owner: Node, spawn_with_projectile_effects: Array, parent_transform: Transform2D):
+	# Execute any spawn modification effects for the projectile
+	print("spawned!")
+	#projectile_effects = spawn_with_projectile_effects
+	var local_transform = parent_transform
+	for effect in spawn_with_projectile_effects:
+		var instance = effect.instantiate()
+		projectile_effects.append(instance)
+		add_child(instance)
+		instance.set_owner(self)
+		instance.modify_creation(owner, spawn_with_projectile_effects, local_transform)
+	# Spawn the default projectile
+	# TODO:: Projectile spawn modification may want to prevent the default from spawning
+	#    how do?
+	owner.add_child(self)
+	projectile_effects = projectile_effects
+	transform = local_transform
+
+func scale_projectile(new_scale: Vector2):
+	$Sprite2D.scale = new_scale
+	$CollisionShape2D.scale = new_scale
+	scale = new_scale
+
+func _ready():
+	if scale != Vector2.ONE:
```

Comment:
:wine_glass: :cheese: 

This is a hack to allow the "scale" you set on the Projectile node itself to impact the actual scale of the base projectile.
Without this, setting "Transform -> Scale" on the base projectile node will change the size of the projectile in the 2D visual UI, but at runtime it will be completely ignored. See [this godot pr](https://github.com/godotengine/godot/issues/5734), also linked in the commit message directly.